app/components/progress_bar/index.tsx

```diff
@@ -1,67 +1,160 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React, {useCallback, useEffect, useState} from 'react';
-import {type LayoutChangeEvent, StyleSheet, type StyleProp, View, type ViewStyle} from 'react-native';
-import Animated, {useAnimatedStyle, useSharedValue, withTiming} from 'react-native-reanimated';
+import React, {useCallback, useEffect} from 'react';
+import {type LayoutChangeEvent, type StyleProp, View, type ViewStyle, StyleSheet} from 'react-native';
+import {Gesture, GestureDetector} from 'react-native-gesture-handler';
+import Animated, {clamp, runOnJS, useAnimatedStyle, useSharedValue, withTiming} from 'react-native-reanimated';
+
+import {useTheme} from '@context/theme';
+import {changeOpacity} from '@utils/theme';
 
 type ProgressBarProps = {
     color: string;
     progress: number;
+    withCursor?: boolean;
     style?: StyleProp<ViewStyle>;
+    onSeek?: (position: number) => void;
 }
 
 const styles = StyleSheet.create({
     container: {
+        position: 'relative',
+        justifyContent: 'center',
+        paddingVertical: 10,
+    },
+    progressBarContainer: {
         height: 4,
         borderRadius: 2,
-        backgroundColor: 'rgba(255, 255, 255, 0.16)',
         overflow: 'hidden',
         width: '100%',
     },
     progressBar: {
         flex: 1,
     },
+    cursor: {
+        position: 'absolute',
+        borderRadius: 100,
+        width: 15,
+        height: 15,
+    },
 });
 
-const ProgressBar = ({color, progress, style}: ProgressBarProps) => {
-    const [width, setWidth] = useState(0);
+const START_CURSOR_VALUE = 0;
+const END_CURSOR_VALUE = 1;
 
+const ProgressBar = ({color, progress, withCursor, style, onSeek}: ProgressBarProps) => {
+    const theme = useTheme();
+    const widthValue = useSharedValue(0);
     const progressValue = useSharedValue(progress);
+    const isGestureActive = useSharedValue(false);
+
+    // eslint-disable-next-line new-cap
```

Comment:
Why do we need this lint disable?

Reply:
I believe this was discussed before.

The eslint rules for the Mattermost Mobile project doesn't allow uppercase letters to be used as constructors as you can see:
![image](https://github.com/user-attachments/assets/9af19311-56b6-448c-ac98-f15d12b0c5bc)

But there is no way around in this case, since this is the pattern for `Gesture.Pan()` there is other places on the code that does the same thing for the Gesture.

Reply:
Sorry if it was discussed before, and thank you for the reminder :smile: